subsys/dfu/fmfu_fdev/cddl/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Output directories inside build dir
+set(src_out ${ZEPHYR_BINARY_DIR}/source/generated)
+set(include_out ${ZEPHYR_BINARY_DIR}/include/generated)
+
+# Make sure that output directory for *.c files exist
+file(MAKE_DIRECTORY ${src_out})
+
+# This file is used as the source for the parser generator
+set(cddl_file ${CMAKE_CURRENT_LIST_DIR}/modem_update.cddl)
+
+# These are the entry types needed by the source code
+set(entry_types Wrapper Sig_structure1 Segments)
+
+set(decode_c_name modem_update_decode.c)
+set(decode_h_name modem_update_decode.h)
+set(types_h_name modem_update_types.h)
+set(decode_c ${src_out}/${decode_c_name})
+set(decode_h ${include_out}/${decode_h_name})
+set(types_h ${include_out}/${types_h_name})
+set(src_install_dir ${NRF_DIR}/subsys/dfu/fmfu_fdev/src)
+set(include_install_dir ${NRF_DIR}/subsys/dfu/fmfu_fdev/include)
+set(license ${CMAKE_CURRENT_LIST_DIR}/license.txt)
+
+function(cat IN_FILE OUT_FILE)
+  file(READ ${IN_FILE} CONTENTS)
+  file(APPEND ${OUT_FILE} "${CONTENTS}")
+endfunction()
+
+foreach(target decode_c decode_h types_h)
+  list(APPEND CAT_COMMANDS COMMAND cat ${license} ${${target}} > ${${target}})
+  list(APPEND CF_COMMANDS COMMAND clang-format -i ${${target}})
+endforeach()
+
+add_custom_command(
+  OUTPUT ${decode_c} ${decode_h} ${types_h}
+  COMMAND
+  ${PYTHON_EXECUTABLE}
+  ${ZEPHYR_CDDL_GEN_MODULE_DIR}/scripts/cddl_gen.py
+  -i ${cddl_file}
+  --oc ${decode_c}
+  --oh ${decode_h}
+  --oht ${types_h}
+  -t ${entry_types}
+  -d # Decode
+  --default-maxq 16 # Up to 16 segments is supported.
+  ${CAT_COMMANDS}
+  ${CF_COMMANDS}
+  COMMENT
+  "Re-generating files based on ${cddl_file}"
+  DEPENDS ${license}
+  )
+
+zephyr_library()
+zephyr_library_sources(${decode_c})
+zephyr_include_directories(${include_out})
+
+# Create install target which allows the user to 'install' the generated
+# parser files into the working tree.
+list(APPEND INSTALL_COMMANDS
```

Comment:
what is the value of creating a list to use, and then using it in `add_custom_target`.

This just makes it harder to see what will actually be executed in that target, as I now have to find the list to read the commands without any gain.

This is different that the earlier case at L35-L38, where you actually were creating the list in a loop, which is not possible in the CMake function directly.